Feature/cu e91cdb better getting started with open service brokers development material tutorial (#335)
meshcloud/meshcloud-docs: docs/meshstack.meshmarketplace.broker-tutorial.md

- **Fetching the catalog of backing services that a service broker offers**
The catalog describes all of the services that can be provisioned through a service broker, and  each service is made up of plans. Plans typically represent the costs and benefits for a given variant of the service. Many services use plans to ‘T-Shirt size’ the service offering (such as “small”, “medium”, and “large”).
- **Provisioning new service instances**
A service instance is a provisioned instance of a service and plan as described in the service broker’s catalog.
- **Connecting and disconnecting applications and containers from those service instances(Bind/Unbind)**
Once a service instance is provisioned, you’ll want your application or container to start communicating with that instance. From a service broker’s perspective, this is called a service binding.
- **Deprovisioning service instances**
This action simply deletes all the resources created upon the initial provisioning of the service instance.

## Quickstarts

To get started with existing projects.

> - We provide an example implementation of a [Service broker](https://github.com/meshcloud/unipipe-service-broker), that can provide you some guidance for your own implementation.Uses git repository for management. The documentation to develope your own service broker in java can be found here [Open Service Broker API in Java](https://docs.spring.io/spring-cloud-open-service-broker/docs/3.2.0/reference/#introduction). Experimental CI/CD pipelines to provision services using [`concourse-ci`](https://github.com/Meshcloud/example-osb-ci) &  [`github actions`](https://github.com/meshcloud/unipipe-osb-cicd-github-actions).

Reviewer: ```suggestion
- We provide an example implementation of a Service broker called [Unipipe Service Broker](https://github.com/meshcloud/unipipe-service-broker). It connects to a git repository that holds the catalog. The git repository is also used to store information about services instances and their bindings. Unipipe service broker relies on a CI/CD pipeline for managing service instances. Experimental CI/CD pipelines that integrate with Unipipe Service Broker using [`concourse-ci`](https://github.com/Meshcloud/example-osb-ci) &  [`github actions`](https://github.com/meshcloud/unipipe-osb-cicd-github-actions).
```

A go library that provides the REST API implementation for the OSB API. Users
implement an interface that uses the types from the
[go-open-service-broker-client](https://github.com/pmorie/go-open-service-broker-client).

- [Cloud service broker](https://github.com/pivotal/cloud-service-broker/):
This service broker uses Terraform to provision and bind services.

***.NET***

- [Open Service Broker API for .NET](https://github.com/AXOOM/OpenServiceBroker):
.NET libraries for client and server implementations of the Open Service Broker API. The client library allows you to call Service Brokers that implement the API using idiomatic C# interfaces and type-safe DTOs. The server library implements the API for you using ASP.NET Core. You simply need to provide implementations for a few interfaces, shielded from the HTTP-related details.

***Python***

- [Python package](https://pypi.org/project/openbrokerapi/): A Python package for building Service Brokers supporting API version 2.13+.

## Other Libraries

[go-open-service-broker-client](https://github.com/pmorie/go-open-service-broker-client):
This library is a golang client for communicating with service brokers,

> useful for Platform developers.

Reviewer: This should go to the bottom of the page where "Open Service Broker Client App" Chapter is.

## Implementation Recommendations

This section has some implementation recommendations based on experience:

> - Follow the [Open Service Broker API (2.16)](https://github.com/openservicebrokerapi/servicebroker/blob/v2.16/spec.md) specification closely.

Reviewer: ```suggestion
- Follow the [Open Service Broker API](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md) specification closely.
```

- CLI that interacts with any Open Service Broker API [eden](https://starkandwayne.com/blog/welcome-to-eden-a-cli-for-every-open-service-broker-api/). The eden CLI is useful for users of a service broker, and also for developers' own dev/test.

## Service owner responsibility

- Define and manage the catalog. Service Broker authors are expected to be cautious when removing Service Offerings and Service Plans from their catalogs, as Platforms might have provisioned Service Instances of these Service Plans. For example, Platforms might restrict the actions that users can perform on existing Service Instances if the associated Service Offering or Service Plan is deleted. Consider your deprecation strategy.

- Service broker authors should also be aware of all the possible scenarios to maintain the service broker and should provide extra endpoints to handle erroneous states of the service broker. Some example scenarios can be found [here](https://github.com/swisscom/open-service-broker) under admin actions section.

> ## OSBAPI Complaint Products

Reviewer: ```suggestion
## OSBAPI Compliant Products
```
